File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/mad/Java/Python/s041542819_commented.py
import sys

class LightScanner:
    def __init__(self, input_stream):
        self.reader = input_stream
        self.buffer = []
        self.index = 0

    def _read(self):
        line = self.reader.readline().strip()
        if line:
            self.buffer.extend(line.split())
        else:
            self.buffer = []

    def string(self):
        if self.index >= len(self.buffer):
            self._read()
        if self.index < len(self.buffer):
            result = self.buffer[self.index]
            self.index += 1
            return result
        else:
            raise EOFError

    def ints(self):
        return int(self.string())

    def longs(self):
        return int(self.string())  # In Python, int can handle long integers

class LightWriter:
    def __init__(self, output_stream):
        self.out = output_stream

    def print(self, s):
        self.out.write(str(s))

    def ans(self, s):
        self.print(s)

    def ansln(self, *n):
        for n1 in n:
            self.ans(n1)
            self.ln()

    def ln(self):
        self.out.write('\n')

    def close(self):
        self.out.flush()

class IntMath:
    @staticmethod
    def min(*v):
        return min(v)

class ArrayUtil:
    @staticmethod
    def lowerBound(a, t, min_index=0):
        max_index = len(a)
        while min_index < max_index:
            mid = (min_index + max_index) // 2
            if t <= a[mid]:
                max_index = mid
            else:
                min_index = mid + 1
        return min_index

class DLazyFaith:
    def solve(self, test_number, in_stream, out_stream):
        a = in_stream.ints()
        b = in_stream.ints()
        q = in_stream.ints()
        s = [-10_000_000_000] * (a + 2)
        t = [-10_000_000_000] * (b + 2)
        for i in range(a):
            s[i + 1] = in_stream.longs()
        for i in range(b):
            t[i + 1] = in_stream.longs()
        s[a + 1] = t[b + 1] = 20_000_000_000
        for _ in range(q):
            x = in_stream.longs()
            sl = x - s[max(0, ArrayUtil.lowerBound(s, x + 1) - 1)]
            tl = x - t[max(0, ArrayUtil.lowerBound(t, x + 1) - 1)]
            sr = s[ArrayUtil.lowerBound(s, x)] - x
            tr = t[ArrayUtil.lowerBound(t, x)] - x
            result = IntMath.min(
                max(sl, tl),
                max(sr, tr),
                2 * sl + tr,
                2 * tl + sr,
                sl + 2 * tr,
                tl + 2 * sr
            )
            out_stream.ansln(result)

def main():
    in_stream = LightScanner(sys.stdin)
    out_stream = LightWriter(sys.stdout)
    solver = DLazyFaith()
    solver.solve(1, in_stream, out_stream)
    out_stream.close()

if __name__ == "__main__":
    main()
# End of Code

File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/mad/Java/Go/s041542819_commented.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

type LightScanner struct {
	reader    *bufio.Reader
	tokenizer []string
}

func NewLightScanner(r *bufio.Reader) *LightScanner {
	return &LightScanner{reader: r}
}

func (ls *LightScanner) string() string {
	if len(ls.tokenizer) == 0 {
		line, _, err := ls.reader.ReadLine()
		if err != nil {
			panic(err)
		}
		ls.tokenizer = strings.Fields(string(line))
	}
	token := ls.tokenizer[0]
	ls.tokenizer = ls.tokenizer[1:]
	return token
}

func (ls *LightScanner) ints() int {
	i, err := strconv.Atoi(ls.string())
	if err != nil {
		panic(err)
	}
	return i
}

func (ls *LightScanner) longs() int64 {
	i, err := strconv.ParseInt(ls.string(), 10, 64)
	if err != nil {
		panic(err)
	}
	return i
}

type LightWriter struct {
	writer *bufio.Writer
}

func NewLightWriter(w *bufio.Writer) *LightWriter {
	return &LightWriter{writer: w}
}

func (lw *LightWriter) print(s string) {
	_, err := lw.writer.WriteString(s)
	if err != nil {
		panic(err)
	}
}

func (lw *LightWriter) ans(s string) {
	lw.print(s)
}

func (lw *LightWriter) ansln(n ...int64) {
	for _, num := range n {
		lw.ans(fmt.Sprintf("%d", num))
		lw.ln()
	}
}

func (lw *LightWriter) ln() {
	lw.print("\n")
}

func (lw *LightWriter) close() {
	err := lw.writer.Flush()
	if err != nil {
		panic(err)
	}
}

func min(values ...int64) int64 {
	minValue := values[0]
	for _, v := range values {
		if v < minValue {
			minValue = v
		}
	}
	return minValue
}

func max(a, b int64) int64 {
	if a > b {
		return a
	}
	return b
}

func lowerBound(a []int64, t int64) int {
	min, max := 0, len(a)
	for min < max {
		mid := (min + max) / 2
		if t <= a[mid] {
			max = mid
		} else {
			min = mid + 1
		}
	}
	return min
}

type DLazyFaith struct{}

func (d *DLazyFaith) solve(testNumber int, in *LightScanner, out *LightWriter) {
	a, b, q := in.ints(), in.ints(), in.ints()
	s := make([]int64, a+2)
	t := make([]int64, b+2)
	s[0], t[0] = -10000000000, -10000000000
	for i := 0; i < a; i++ {
		s[i+1] = in.longs()
	}
	for i := 0; i < b; i++ {
		t[i+1] = in.longs()
	}
	s[a+1], t[b+1] = 20000000000, 20000000000
	for i := 0; i < q; i++ {
		x := in.longs()
		sl := x - s[max(0, int64(lowerBound(s, x+1)-1))]
		tl := x - t[max(0, int64(lowerBound(t, x+1)-1))]
		sr := s[lowerBound(s, x)] - x
		tr := t[lowerBound(t, x)] - x
		out.ansln(min(
			max(sl, tl),
			max(sr, tr),
			2*sl+tr,
			2*tl+sr,
			sl+2*tr,
			tl+2*sr,
		))
	}
}

func main() {
	in := NewLightScanner(bufio.NewReader(os.Stdin))
	out := NewLightWriter(bufio.NewWriter(os.Stdout))
	defer out.close()
	solver := &DLazyFaith{}
	solver.solve(1, in, out)
}

// End of Code
